docs/testplan/HA-SmartSwitch-test-plan.md

```diff
@@ -0,0 +1,171 @@
+      - [Module 3 	HA state pinned by upstream service](#module-3--ha-state-pinned-by-upstream-service)
+      - [Module 4 	Link Failures](#module-4--link-failures)
+      - [Module 5 	Critical Process Crash](#module-5--critical-process-crash)
+      - [Module 6 	Power down and hardware failure](#module-6--power-down-and-hardware-failure)
+      - [Module 7 	Operations](#module-7--operations)
+    - [Upstream (T0->T2) Traffic Verification](#upstream-t0t2-traffic-verification)
+      - [Module 1 Normal OP](#module-1-normal-op)
+  - [Test Utilities](#test-utilities)
+
+## Scope 
+This document proposes solutions for Smart Switch High-Availability test plans. The document will cover smart switch test scenarios.
+
+The goal of this test plan is to verify HA state machine behavior in normal operation scenarios and network failure scenarios with [t1-smartswitch-ha](https://github.com/sonic-net/sonic-mgmt/blob/master/ansible/vars/topo_t1-smartswitch-ha.yml) topology. Both control plane and data plane will need to be verified in the test cases. 
+
+## Topology
+
+Tests will run on a sonic-mgmt testbed. The key components will be:
+1. Test server
+1. Fanout switches
+    * root fanout switch
+    * leaf fanout switch
+1. SONiC DUTs -- the two smartswitches.
+
+Key aspects of the physical connection:
+1. Every DUT port is connected to the leaf fanout switch.
+1. Every leaf fanout switch has unique VLAN tag for every DUT port.
+1. Root fanout switch connects leaf fanout switches and test servers using 802.1Q trunks. 
+
+Any test server can access any DUT port by sending a packet with the port VLAN tag. In test servers, a set of dockers can be created for running cEOS to simulate neighbors of the SONiC DUT. A PTF container (based on the PTF test framework) for injecting/sniffing packets will be created. The PTF docker, the cEOS dockers and VLAN interfaces in test server can be interconnected by open vSwitch bridges. 
+
+![physical-connection](./Img/topo_smartswitch_ha.png)
+
+Please refer to [sonic-mgmt testbed overview](https://github.com/sonic-net/sonic-mgmt/blob/master/docs/testbed/README.testbed.Overview.md) for more about testbed setup and diagrams. 
+
+
+## Test Methodology
+
+1. Open flow rules are configured on OVS Bridges to control the traffic flow:
+    * All packets sent out from Neighbor are only forwarded to DUT
+    * All packets sent out from DUT are forwarded to Neighbor and PTF
+    * All packets sent out from PTF are only forwarded to DUT
+1. Traffic will be sent from PTF docker, and sniffed on PTF docker by sonic-mgmt test scripts.
+1. Depending on the test cases, switchovers will be triggered and verified. 
+1. Control plane status, DPU counters, metering data will be measured for the tests.
+
+## Test Plan
+Assuming there is a pair of DPUs in the system, and at the step of test setup, we have DPU-1 as active, DPU-2 as standby. 
+
+
+### Downstream (T2->T0) Traffic Verification 
```

Comment:
i probably kinda forgot why we are checking t2->t0 explicitly, but it should be injecting from t2 and receive from t2 (after HA inline sync).

Reply:
maybe better to lift the submodule up

Reply:
> i probably kinda forgot why we are checking t2->t0 explicitly, but it should be injecting from t2 and receive from t2 (after HA inline sync).

Are we talking about control plane or data plane here?

Reply:
As discussed offline, addressed.